src/quote/swap/interfaces.ts

```diff
@@ -0,0 +1,19 @@
+import { ZeroExApiSwapResponse } from 'quote/swap/adapters'
+
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+interface SwapQuote extends ZeroExApiSwapResponse {}
+export interface SwapQuoteRequest {
+  chainId: number
+  inputToken: string
+  outputToken: string
+  // Either one of these must be set
+  inputAmount?: string
+  outputAmount?: string
+  // Optional
+  includedSources?: string // TODO:
+  slippage?: number
+}
+
+export interface SwapQuoteProvider {
+  getSwapQuote(request: SwapQuoteRequest): Promise<SwapQuote | null>
+}
```

Comment:
The new protocol that should be adapter by any SwapQuoteProvider. I went still naming it ` getSwapQuote` as I think it's good differentation from the SDK's `getQuote` functions.